### DISCUSSION TITLE
License Question - I changed the license on my forked repo

### DISCUSSION THREAD

**Discussion by thedoggybrad** [uBlockOrigin/uAssets#13261] [Q&A]
I forked this repo. The filters on my repo are a little bit modified but still rely on this filter and even set up a GitHub action to sync it. I changed the license to ISC, which is located here https://github.com/thedoggybrad/UBlockFiltersforall/blob/master/LICENSE. Is it okay or should I revert it back to the original license of this repo?

**Comment by thedoggybrad** [comment]


**Reply by okiehsch** [reply]
~~Sure, you can fork software with a GPL license and change it to a MIT license, on which your ISC license is based on.~~

**Reply by thedoggybrad** [reply]
Thank You for answering my question.

**Reply by MasterKia** [reply]
> Sure, you can fork software with a GPL license and change it to a MIT license, on which your ISC license is based on.

You _**can't change**_ the license of GPL software, GPL is a copyleft license which means any forked version or modification should carry the same license (_GPL_).

**Reply by okiehsch** [reply]
You can't "change" the license but you can redistribute a MIT license under a new license. Which was  what the OP intended to do with a GPL license as far as I understood.
**However after reading through it @MasterKia is correct you can't even redistribute a GPL license.**

Sorry about my faulty advice.

**Comment by MasterKia** [comment]
Notice the "**Same license**" condition.
https://github.com/uBlockOrigin/uAssets/blob/master/LICENSE:
![image](https://user-images.githubusercontent.com/17685483/171989427-cccfd0bb-ae05-4e83-8db2-8b0acfd58b9d.png)

**Comment by MasterKia** [comment]
@thedoggybrad
On a side note and food for thought, the GPL isn't a "restrictive" license; the conditions of a copyleft license like GPL is to ensure _nobody_ can take away the freedoms of users, unlike what (in my opinion) flawed licenses such as BSD or MIT allow (like what Apple is doing with the BSD operating system, they take code but don't contribute anything back).

**Comment by jsteave** [comment]
@[thedoggybrad](https://pakipackages.com/jazz/jazz-call-packages/) Hi,
First of all: thanks for fastpages! I started trying it out over the holidays and am really impressed with both the capabilities of the system and how the github template you’ve created makes things. I was dreading having to spend a day or so shaving a yak to get started, but everything pretty much just worked and now I have a great-looking blog to add to… nice! :smiley:

I do have a question about licensing though: the fastpages template sets up a repo with an Apache license. I have no objections to the license itself, but prefer that the blog itself be licensed using CC-BY-4.0. I think that’s more appropriate for the contents of a blog than a software license like Apache, though I recognize that things are a bit tricky with a fastpages repo, which incorporates both code and content.

I’ve added a footer to the template to add the appropriate license grant to the published blog posts, but I’d also like to make the licensing clear in the github repo itself. The easiest way for me to do this would be to just replace the top level LICENSE file, but I definitely want to ask before doing that. Another option - less obvious for people browsing in github - would be for me to add separate LICENSE files to the _notebooks, _posts, and _word directories.

Any suggestions on a good way to do this that doesn’t step on the fastpages team’s collective toes and makes sure that credit is given?

Thanks,
